src/main/java/co/cask/hydrator/salesforce/SObjectsDescribeResult.java

```diff
@@ -97,6 +98,7 @@ private void addSObjectDescribe(DescribeSObjectResult sObjectDescribe) {
         Function.identity(),
         (o, n) -> n,
         LinkedHashMap::new)); // preserve field order for queries by sObject
+    fields = new CaseInsensitiveMap(fields);
```

Comment:
1. do we really need new dependency only to avoid toLowerCase transformation on put/get?
2. how did you make sure it does not violate LinkedHashMap order?
3. you don't have toLowerCase tranformation for map in constructor